File: src/Shopperior.Api/Shopperior.WebApi/ShoppingLists/Models/ListItemDto.cs
namespace Shopperior.WebApi.ShoppingLists.Models;

public class ListItemDto
{
    public Guid Guid { get; set; }
    public Guid ShoppingListGuid { get; set; }
    public string Name { get; set; }
    public string Brand { get; set; }
    public string Comment { get; set; }
    public double Quantity { get; set; }
    public string Measurement { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal TotalPrice { get; set; }
    public bool IsInCart { get; set; }
    public DateTime? EnteredCartTime { get; set; }
    public bool HasPurchased { get; set; }
    public DateTime? PurchasedTime { get; set; }
}

File: src/shopperior-ng/src/app/core/data/shopperior-api/models/list-item-dto.ts
export interface ListItemDto {
  guid: string;
  shoppingListGuid: string;
  categoryGuid: string;
  name: string;
  brand: string;
  comment: string;
  quantity: number;
  measurement: string;
  unitPrice: number;
  totalPrice: number;
  isInCart: boolean;
  hasPurchased: boolean;
}